pkg/tfvar/tfvar.go

```diff
@@ -110,7 +114,60 @@ func WriteAsTFVars(w io.Writer, vars []Variable) error {
 	_, err := f.WriteTo(w)
 	return errors.Wrap(err, "tfvar: failed to write as tfvars")
 }
+func WriteAsWorkspacePayload(w io.Writer, vars []Variable) error {
+	var payload error
 
+	for _, v := range vars {
+		val := convertNull(v.Value)
+
+		t := hclwrite.TokensForValue(val)
+		t = oneliner(t)
+		b := hclwrite.Format(t.Bytes())
```

Comment:
from the docs:

```
// Format takes source code and performs simple whitespace changes to transform
// it to a canonical layout style
```

Is Format what's wanted here? It looks like you're trying to turn the value `val` into a string representation, is that right? Is that possible for all Variables?

I think these HCL methods are mostly for writing to an HCL file, so they deal mostly with whitespace padding.

It looks like the ctypes themselves support json marshalling. see https://pkg.go.dev/github.com/zclconf/go-cty@v1.10.0/cty/json#Marshal

Is it possible to just have the value be a cty.Value, and let the json marshaller take care of the rest?

i.e. remove lines 123-128 and instead do this (as in the example below)

```
jdata, _ := json.Marshal(
		map[string]interface{}{
			"data": struct {
				Type       string `json:"type"`
				Attributes map[string]interface{} `json:"attributes"`
			}{
				"vars",
				map[string]interface{}{
					"value":  val, // let json marshaller deal with formatting
				},
			},
		},
	)
```

Reply:
Interesting, I think I was using Format the same way in WriteAsEnvVars to construct the value which I assume the format of values is the same here. (I don't have Terraform Enterprise to test this)

> Is it possible to just have the value be a cty.Value, and let the json marshaller take care of the rest?

Would be interested in finding out if the above works.

Reply:
I have attempted to get the json.Marshall method to work, again it's just not working for me, and every change then means a suggested another change, and i don't totally understand the language to follow that chain. 

i believe it needs the json var declaring first but that suggested it was incorrect 
and returning the json object required an additional function. 

i'd be interested to see you follow up pull